### PR TITLE
empty list cannot be cleared issue fixed.

### PR DESCRIPTION
**Pull request by hsali**
## Description ##
checking list before resetting it.

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage
- [ ] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ]  check list and clear

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

**Comment by anirudhacharya** [comment]
@mxnet-label-bot add [pr-awaiting-review]

File: example/ctc/multiproc_data.py [hunk 1]
```diff
@@ -141,4 +141,5 @@ def reset(self):
         print("Queue size on reset: {}".format(qsize))
         for i, p in enumerate(self.proc):
             p.join()
-        self.proc.clear()
+        if len(self.proc) > 0:
```
Comment (vandanavk) [inline example/ctc/multiproc_data.py]: maybe a more Pythonic way would be `if self.proc` ?

**Comment by karan6181** [comment]
@hsali Could you please address the review comments? Thanks!

File: example/ctc/multiproc_data.py (hunk 1 above)
Comment (hsali) [reply example/ctc/multiproc_data.py]: I was using python2.7. I faced issue at that time. that is also good suggestion.

**Comment by piyushghai** [comment]
@hsali Gentle ping....

**Comment by vandanavk** [comment]
LGTM.

**Comment by karan6181** [comment]
@hsali Thanks for your contribution. Could you please address the review comments and Is this PR good to go for merge?

**Comment by piyushghai** [comment]
Can a committer merge this PR ? This is good to go!

**Comment by szha** [comment]
@mxnet-bot run ci [all]

**Comment by szha** [comment]
sorry that it took (quite) a while. thanks for the fix @hsali

**Pull request merged**